preup/script_api.py

```diff
@@ -532,6 +532,12 @@ def print_home_dirs(user_name=""):
         pass
 
 
+def add_pkg_to_kickstart(pkg_name):
+    for pkg in pkg_name.split():
+        write_to_file(SPECIAL_PKG_LIST, "a+b", pkg)
+    return 0
+
+
```

Comment:
what about:
```python
def add_pkg_to_kickstart(pkgs):
    empty = False
    if(!isinstanceof(pkgs, list)):
        # string - pkgs delimited by whitespace
        if(len(pkgs.strip()) == 0):
            empty = True
        else:
           # make list from string
            pkgs = pkgs.strip().split()
    else:
        # list of packages
        if(len(pkgs) == 0):
            empty = True
    if(empty is True):
        log_debug("Missing parameters! Any package will be added.")
        return 1
    write_to_file(SPECIAL_PKG_LIST, "a+b", pkgs)
    return 0
```

It accepts list or string. I count that developer will use more list than strings in this case. Function write_to_file accept list & string too. It's untested, so maybe there are mistakes. Just written here as comment.